src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,32 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CREDS_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login cannot be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password cannot be null");
+        }
+        if (user.getLogin().length() < MIN_CREDS_LENGTH) {
+            throw new InvalidDataException("Login must be at least "
+                    + MIN_CREDS_LENGTH + " characters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User already exists");
+        }
+        if (user.getPassword().length() < MIN_CREDS_LENGTH) {
+            throw new InvalidDataException("Password must be at least "
+                    + MIN_CREDS_LENGTH + " characters");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE) {
```

Comment:
The check `user.getAge() == null || user.getAge() < MIN_AGE` is correct for ensuring the age is not null and meets the minimum age requirement. However, ensure that the `User` class has a method `getAge()` that returns an `Integer` to handle null values appropriately.